api/controllers/ApplicationTypeController.js

```
@@ -54,93 +55,86 @@ module.exports = {
-                        // set initial submit status to false, meaning it application has not yet been submitted
-                        req.session.appSubmittedStatus = false;
-                        req.session.email_sent = false;
-
-                        return res.view(serviceSelectorView, {
-                            application_id: 0,
-                            userServiceURL:
-                                sails.config.customURLs.userServiceURL,
-                            error_report: false,
-                            changing: false,
-                            form_values: false,
-                            submit_status: req.session.appSubmittedStatus,
-                            current_uri: req.originalUrl,
-                            user_data: userData,
-                            back_link: req.session.startBackLink,
-                            //disableStandardServiceSection: disableStandardServiceSection
-                        });
-                    });
-                });
-            });
-
-        }
-
         req.session.appId = false; // reset the appId so a new session is used
         // set initial submit status to false, meaning it application has not yet been submitted
         req.session.appSubmittedStatus = false;
 
-        return res.view('applicationForms/applicationType.ejs', {
+        const userModels = getUserModels(req._sails.config.userServiceSequelize);
+
+        return this._renderServiceSelectionPage(req, res, userModels);
+    },
+
+    _renderServiceSelectionPage(req, res, userModels) {
+        const userLoggedIn = HelperService.LoggedInStatus(req);
+        const userData = HelperService.getUserData(req, res);
+        const errorReport = req.flash('serviceSelectError')[0];
+
+        const pageData = {
             application_id: 0,
             userServiceURL: sails.config.customURLs.userServiceURL,
-            error_report: false,
+            error_report: Boolean(errorReport),
```

`error_report: req.flash('serviceSelectError')` ideally should return the error message "You must select a service type." - at the moment it appears that there can only be one error, however by passing the string through it makes it much more flexible